docs/node/workflows.md

```diff
@@ -65,54 +65,81 @@ There is no official support for querying the state of asynchronously started Wo
+  - Equivalent to `new CancellationScope().run(fn)`.
+- [`nonCancellable(fn)`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#noncancellable): prevents cancellation from propagating to children.
+  - Equivalent to `new CancellationScope({ cancellable: false }).run(fn)`.
+- [`withTimeout(timeoutMs, fn)`](https://nodejs.temporal.io/api/classes/workflow.cancellationscope#withtimeout): If timeout triggers before all activities complete, the Workflow will fail with a `CancelledError`.
+  - Equivalent to `new CancellationScope({ cancellable: true, timeout: timeoutMs }).run(fn)`.
+
+Cancellation propagates from outer scopes to inner ones and is handled by catching `CancelledError`s thrown by cancellable operations (see below).
+
+Scopes all return native JS Promises, so you can use all the familiar Promise APIs like `Promise.all` and `Promise.race` to model your async logic.
+Other APIs you can use:
+
+- `CancellationScope.current()`: get the current scope
+- `scope.cancel()`: cancel all operations inside a `scope`
+- `scope.run()`: run an async function within a `scope`
+- `scope.cancelRequested()`: a promise that resolves when a cancel request hits a `nonCancellable` workflow
 
 When a `CancellationScope` is cancelled, it propagates cancellation in any child scopes and of any cancellable operations created within it, such as:
 
 - Activities
 - Timers (created with the [`sleep`](https://nodejs.temporal.io/api/modules/workflow#sleep) function)
 - [`Trigger`](https://nodejs.temporal.io/api/classes/workflow.trigger)s
 
-## Examples
-
-### Cancel a timer from Workflow code
+### Internal Cancellation example
 
 <!--SNIPSTART nodejs-cancel-a-timer-from-workflow-->
 <!--SNIPEND-->
 
-### Alternatively, the preceding can be written as
+Alternatively, the preceding can be written as:
 
 <!--SNIPSTART nodejs-cancel-a-timer-from-workflow-alternative-impl-->
 <!--SNIPEND-->
 
-### Run multiple activities with a single deadline
+### External Cancellation example
 
-<!--SNIPSTART nodejs-multiple-activities-single-timeout-workflow-->
+Handle Workflow cancellation by an external client while an Activity is running:
+
+<!--SNIPSTART nodejs-handle-external-workflow-cancellation-while-activity-running-->
 <!--SNIPEND-->
 
-### `nonCancellable` prevents cancellation from propagating to children
+### `nonCancellable` example
+
+`CancellationScope.nonCancellable` prevents cancellation from propagating to children:
 
 <!--SNIPSTART nodejs-non-cancellable-shields-children-->
 <!--SNIPEND-->
 
-### `cancelRequested` may be awaited upon to make Workflow aware of cancellation while waiting on `nonCancellable` scopes
+### `withTimeout` example
+
+A very common operation is to cancel one or more activities if a deadline elapses:
+
+<!--SNIPSTART nodejs-multiple-activities-single-timeout-workflow-->
+<!--SNIPEND-->
+
+### `scope.cancelRequested`
+
+You can await `cancelRequested` to make Workflow aware of cancellation while waiting on `nonCancellable` scopes:
```

Comment:
Not workflow, but the specific scope. 

@bergundy should we even call it nonCancellable? I believe it is cancellable as it can be canceled using its handle. It is "disconnected" as the cancellation is not propagated from the parent in an automated manner.

Reply:
"Workflow" means the workflow code here, anything that runs in a `nonCancellable` scope will not be cancelled when the scope is cancelled, although the `cancelRequested` Promise will still `reject`ed  in case you want to know if there was a cancellation request.
Take a look at the sample it should make that clear.